libdnf/base/transaction.cpp

```diff
@@ -277,33 +246,9 @@ void Transaction::Impl::set_transaction(rpm::solv::GoalPrivate & solved_goal, Go
             make_transaction_package(id, TransactionPackage::Action::DOWNGRADE, solved_goal, replaced));
     }
 
-    auto list_removes = solved_goal.list_removes();
-    if (!list_removes.empty()) {
-        rpm::PackageQuery remaining_installed(base, rpm::PackageQuery::ExcludeFlags::IGNORE_EXCLUDES);
-        remaining_installed.filter_installed();
-        for (auto id : list_removes) {
-            remaining_installed.p_impl->remove(id);
-        }
-        rpm::PackageSet tmp_set(base);
-
-        // https://bugzilla.redhat.com/show_bug.cgi?id=1921063
-        // To keep a reason of installonly pkgs in DB for remove step it requires TSI with reason change
-        for (auto id : list_removes) {
```

Comment:
I am really happy to see the removal of this code, because it is not the correct place where it should be handled. But I don't see the logic that replaces it. I know that we already discussed possible solution but I feel not quite comfortable because you corrected the code but it creates an issue in functionality.

Reply:
Or did I overlooked the new logic that replaced the functionality?

Reply:
I think I found the new logic in transaction.cpp but I have to check again.

Reply:
It is there therefore I am closing this conversation as resolve. I am sorry for the noise.